pkg/auth/mtls_authhandler.go

```diff
@@ -0,0 +1,289 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package auth
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/auth/certs"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/policy"
+)
+
+type mtlsParams struct {
+	cell.In
+
+	CertificateProviders []certs.CertificateProvider `group:"certificateProviders"`
+}
+
+func newMTLSAuthHandler(lc hive.Lifecycle, cfg MTLSConfig, params mtlsParams, log logrus.FieldLogger) authHandlerResult {
+	if cfg.MTLSListenerPort == 0 {
+		log.Info("mTLS Auth Handler is disabled as no port is configured")
+		return authHandlerResult{}
+	}
+	var cert certs.CertificateProvider
+	for _, c := range params.CertificateProviders {
```

Comment:
it kind of feels wrong to provide the possibility to accept multiple certificate providers but at the end only support one.

if `mtlsParams` would define a single `CertificateProvider` instead of the slice with the `value group` it would already make this validation and fail if multiple cells are providing an implementation.

since this "mtls-authhandler" is an actual "mtls-spiffe-authhandler" (as indicated by `authType` -> `return policy.AuthTypeMTLSSpiffe`) we might want to expect a single `SpiffeCertificateProvider` (or maybe even `CertificateProvider`) in the params.

once we might implement a second certificateprovider implementation (e.g. certmanager) - we need to refactor some parts anyway. until then, we can get rid of the value group.